### DISCUSSION TITLE
CTV 6.1 can open more than 1 istance for ip. why and how to choise/fix?

### DISCUSSION THREAD

**Discussion by miniliano** [KevinBytesTheDust/CVAmp#186] [General]
starting from the fact that I have excellent quality proxies, I found that I can open more than one instance for each proxy, This makes me doubt (I also checked the instance opening code that sends a request to the proxies list) and I arrived to the conclusion that a proxy can open multiple instances!! And if this happens, another question: assuming that I have 10 proxies and spaw 10 instances, how can I be sure that each instance has a different proxy AND INSTEAD THE SAME PROXY in the list IS NOT LOADED TWO OR MORE TIMES?
Is there a way to check which proxy an instance is using and if there are doubles or triples? I think it's very important. Personally, now I have done educational tests with 50 proxies and I could even open 100 instances if I wanted.
I kindly ask for clarification on this opening mechanism and how proxies are chosen from the list. thank you very much and congratulations for the excellent work

**Comment by jlplenio** [comment]
Hey @miniliano. You are correct that, depending on the quality of your proxy, multiple instances can use the same proxy and be counted as unique viewers. Currently, the proxies in the list are used in circle, so with 5 proxies and 6 instances, it should go 1,2,3,4,5,1 and so on.
https://github.com/jlplenio/crude-twitch-viewer-bot/blob/8380c49fdeba680c202b124c789c72c7d0abaa03/ctvbot/proxy.py#L52-L53

In the Feature Preview 0.7.0v3, I established a debug-menu with info about the instance. I will add the IP to it here https://github.com/jlplenio/crude-twitch-viewer-bot/discussions/177

Do you have further questions?
